hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirXAttrOp.java

```diff
@@ -326,6 +323,11 @@ static INode unprotectedSetXAttrs(
         throw new IOException("Can only set '" +
             SECURITY_XATTR_UNREADABLE_BY_SUPERUSER + "' on a file.");
       }
+
+      if (xaName.contains(SNAPSHOT_XATTR_NAME)) {
+        Preconditions.checkArgument(inode.isDirectory() &&
```

Comment:
To be consistent with SECURITY_XATTR_UNREADABLE_BY_SUPERUSER, should we throw IOException?

Reply:
Addressed in the current patch.